hub/policy/src/main/java/uk/gov/ida/hub/policy/factories/CacheFactory.java

```diff
@@ -0,0 +1,27 @@
+package uk.gov.ida.hub.policy.factories;
+
+import org.joda.time.DateTime;
+import uk.gov.ida.hub.policy.controllogic.TransitionCache;
+import uk.gov.ida.hub.policy.domain.SessionId;
+import uk.gov.ida.hub.policy.domain.State;
+import uk.gov.ida.shared.dropwizard.infinispan.util.InfinispanCacheManager;
+
+import java.util.Optional;
+import java.util.concurrent.ConcurrentMap;
+
+public class CacheFactory {
+
```

Comment:
We could refactor this so the getStateCache() and getDateTimeCache becomes a single method get(String cachename). We could also replace the strings with an enum to prevent invalid cache names being passed.